internal/sdkv2provider/schema_cloudflare_access_mutual_tls_certificate.go

```diff
@@ -37,7 +48,31 @@ func resourceCloudflareAccessMutualTLSCertificateSchema() map[string]*schema.Sch
 			Elem: &schema.Schema{
 				Type: schema.TypeString,
 			},
-			Description: "The hostnames that will be prompted for this certificate.",
+			Description:           "The hostnames that will be prompted for this certificate.",
+			DiffSuppressOnRefresh: true,
+			DiffSuppressFunc: func(k, oldHostname, newHostname string, d *schema.ResourceData) bool {
+				// never suppress the list count
+				if strings.HasSuffix(k, "#") {
+					return false
+				}
+
+				oldInterface, newInterface := d.GetChange("associated_hostnames")
+				oldHostnames, newHostnames := expandInterfaceToStringList(oldInterface), expandInterfaceToStringList(newInterface)
+				if len(oldHostnames) != len(newHostnames) {
+					// size discrepency (an entry was added or removed)
+					return false
+				}
+
+				for _, hostname := range oldHostnames {
+					if !includes(newHostnames, hostname) {
+						// size is the same, but before and after contain different entries (an entry was renamed)
+						return false
+					}
+				}
```

Comment:
This operation is only performed on the `oldHostnames` slice. I think that since we know their lengths are the same and if old contains something that new doesn't, then that inherently means an entry was renamed and therefore we don't need to perform this operation on `newHostnames`, but I want to call this out to make sure my thinking is correct :P